src/main/resources/sql/deleteDiscreteGroundWaterData.sql

```diff
@@ -1,9 +1,10 @@
 delete
-from discrete_ground_water
-where exists
-    (select null
-        from field_visit_readings_by_loc
-           where json_data_id = ?
-           and partition_number = ?
-           and discrete_ground_water.location_identifier = field_visit_readings_by_loc.location_identifier
-    )
+    from discrete_ground_water
+    where discrete_ground_water_id in
+        (select discrete_ground_water_id
+            from field_visit_readings_by_loc
+            join discrete_ground_water
+                on discrete_ground_water.location_identifier = field_visit_readings_by_loc.location_identifier
+            where json_data_id = ? and
+            partition_number = ?
+        )
```

Comment:
@orosz-usgs I was looking at the doc you posted in the teams wiki - should this be accompanied by indexes on both the `discrete_ground_water.location_identifier` and the `field_visit_readings_by_loc.location_identifier`?

Reply:
Likely an index on discrete_ground_water  location_identifier would help, how much depends on how much the location_identifier divides the data. The more and smaller sets the better. There should be enough (30-40 or
more?), that an index would be beneficial.

In this case, probably do not need to add an index to field_visit_readings_by_loc table. Since we are  selecting by 
json_data_id  and  partition_number and index on those columns should be used. This already filters out any
records from field_visit_readings_by_loc  that we do not want to be part of the join.